sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2034,4 +2034,31 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       }
     }
   }
+
+  test("SPARK-21610: Corrupt records are not handled properly when creating a dataframe " +
+    "from a file") {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+      val writer = new OutputStreamWriter(new FileOutputStream(path), StandardCharsets.UTF_8)
+      val data =
+        """{"field": 1}
+          |{"field": 2}
+          |{"field": "3"}""".stripMargin
+      writer.write(data)
+      writer.close()
+
+      val schema = new StructType()
+        .add("field", ByteType)
+        .add("_corrupt_record", StringType)
+
+      val errMsg = intercept[SparkException] {
+        spark.read
+          .schema(schema)
+          .json(path)
+          .select("_corrupt_record")
+          .show()
+      }.getMessage
+      assert(errMsg.contains("'_corrupt_record' must be selected along with input schema."))
+    }
+  }
```

Comment:
It can be simplified to 

```Scala
    withTempPath { dir =>
      val path = dir.getCanonicalPath
      val data =
        """{"field": 1}
        |{"field": 2}
        |{"field": "3"}""".stripMargin
      sparkContext.parallelize(data, 1).saveAsTextFile(path)

      val e = intercept[SparkException] {
        spark.read.json(path).select("_corrupt_record").collect()
      }.getMessage
      assert(e.contains("'_corrupt_record' must be selected along with input schema."))
    }
```